server/app/app.py

```diff
@@ -53,5 +40,33 @@
 app.add_url_rule("/", endpoint="index")
 
 
+def run(args):
+    global data
+    title = args.title
+    if not title:
+        title = os.path.basename(os.path.normpath(args.data_directory))
+    api_base = f"http://0.0.0.0:{args.port}/api/"
+    app.config.update(
+        DATASET_TITLE=title,
+        CXG_API_BASE=api_base
+    )
+    if args.engine == "scanpy":
+        from .scanpy_engine.scanpy_engine import ScanpyEngine
+        data = ScanpyEngine(args.data_directory, schema="data_schema.json")
+    app.run(host="0.0.0.0", debug=True, port=args.port)
+
+
 def main():
-    app.run(host="0.0.0.0", debug=True, port=5005)
+    parser = argparse.ArgumentParser(description="Cellxgene is a tool for exploring single cell expression.")
+    subparsers = parser.add_subparsers(dest="cellxgene_command")
+    run_subparser = subparsers.add_parser("run", help="run cellxgene")
+    run_subparser.add_argument("--title", "-t", help="Title to display (default = data directory name")
+    run_subparser.add_argument("--port", help="Port to run server on.", type=int, default=5005)
+    run_subparser.add_argument("engine", help="The underlying structure of the data")
```

Comment:
Is this right?

Reply:
Is which right?

Reply:
I thought engine was EM2 vs etc.

Reply:
yeah, I can write a better help text for that. I'm not quite sure how to describe it best in the help text.

Reply:
If you make it a [choice](https://docs.python.org/3/library/argparse.html#choices), it might be pretty self-describing.

Reply:
Shifted from `cellxgene run <engine>` to `cellxgene <engine>` I think that make this a little clearer.